### DISCUSSION TITLE
Skip Reset of Lesser Versions During Increment

### DISCUSSION THREAD

**Discussion by chickdan** [fastlane/fastlane#17887] [Ideas]
When running an `increment_version` command the lesser versions are always reset to 0. It would be nice to have an option, maybe `skip_reset:boolean`, to _keep_ the lesser versions with the exception of the bundle version. 

For example if my app is version 1.2.3(2) and I run: 
`increment_version_minor` I currently expect the output to be 1.3.0(0) 
but if I run the command with the new param:
`increment_version_minor skip_reset:true` I would expect the output to be 1.3.3(0). 

I have built a workaround solution by taking code directly from [increment_version_number.rb](https://github.com/fastlane/fastlane/blob/48151291f2c4949c3b1b9919ba2cc81a7cc33293/fastlane/lib/fastlane/actions/increment_version_number.rb) into my own lane that utilizes the version assignment block around lines 45-65, but it would be nice to have a cleaner option.

**Comment by rogerluan** [comment]
Hey @chickdan 👋 

I can be mistaken here, but I don't think that'd follow [semantic versioning](https://semver.org), which's a good practice. Mind me asking why you'd like to use that approach of versioning? 🙏

**Reply by chickdan** [reply]
You are correct that it would enable someone to not follow semantic versioning, however not all software needs to follow semver. Since Fastlane is largely used for apps it makes sense to support other versioning schemes as well such as release trains. Many companies don't follow semver for their consumer facing products; [JetBrains IDEs](https://blog.jetbrains.com/blog/2016/03/09/jetbrains-toolbox-release-and-versioning-changes/), Sketch, Chrome, and even Android/iOS and that is the purpose of this proposal.
